### DISCUSSION TITLE
Help me understand interval/misfire/concurrent jobs relationship

### DISCUSSION THREAD

**Discussion by robborden** [quartznet/quartznet#370] [General] [answered]
I am running a job repeatedly using .WithSimpleSchedule() .WithInterval().   My job takes anywhere from .5-10 seconds to complete, depending on the amount of buffered workload it needs to do.  The desired behavior is for the process to run singularly, and no matter how long it takes to complete, Quartz should wait the WithInterval timespan (2000ms) and then start the job again.  I have tried to accomplish this by declaring the base class that all my jobs inherit from like this:

> [DisallowConcurrentExecution] 
> public abstract class IMicroserviceJob : IJob

However, the behavior I feel I'm experiencing is that the Job runs at seemingly random intervals "around" the amount of time (within a few seconds) after the job completes.  With a 1000ms timespan, the job often executes several times in one second.  If I set it to 5000ms, the job executes every few seconds, but sometimes every second for several seconds in a row.  If I set it to 10000ms, it seems to run about every 4-8 seconds.  Ultimately, it seems like it is just running an instance "around" every N ms, irrespective of the already running job, kind of like javascript's "setInterval()" function.

Can someone explain to me how Quartz behaves in this scenario, or help me find the docs where this is detailed?

Some specific questions:
Does DisallowConcurrentExecution cause a misfire if a trigger tries to overlap?
Does Quartz fire on an interval irrespective of other job instances that are running, or does the interval start after the current instance finishes running?
Is the above behavior the same when ConcurretExecution is allowed or disallowed?
Are IJob attributes "inherited" attributes?

**Comment by lahma** [comment, marked as answer]
Some answers (and trying to close standing issues):

* misfires happen regardless of DisallowConcurrentExecution 
* intervals are based on triggers, not jobs, so next triggering (or trial of it) happens regardless whether job is running
* should be the same behavior regardless of DisallowConcurrentExecution 
* job attributes are inherited in such way that job inheriting from another class gets same attributes
